examples/features/authz/server/main.go

```diff
@@ -0,0 +1,255 @@
+		return nil, errInvalidToken
+	}
+	m, err := handler(ctx, req)
+	if err != nil {
+		logger("RPC failed with error %v", err)
+	}
+	return m, err
+}
+
+// wrappedStream wraps around the embedded grpc.ServerStream, and intercepts the RecvMsg and
+// SendMsg method call.
+type wrappedStream struct {
+	grpc.ServerStream
+	ctx context.Context
+}
+
+func (w *wrappedStream) Context() context.Context {
+	return w.ctx
+}
+
+func (w *wrappedStream) RecvMsg(m interface{}) error {
+	logger("Receive a message (Type: %T) at %s", m, time.Now().Format(time.RFC3339))
+	return w.ServerStream.RecvMsg(m)
+}
+
+func (w *wrappedStream) SendMsg(m interface{}) error {
+	logger("Send a message (Type: %T) at %v", m, time.Now().Format(time.RFC3339))
+	return w.ServerStream.SendMsg(m)
+}
+
+func newWrappedStream(ctx context.Context, s grpc.ServerStream) grpc.ServerStream {
+	return &wrappedStream{s, ctx}
+}
+
+func authStreamInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	// authentication (token verification)
+	md, ok := metadata.FromIncomingContext(ss.Context())
+	if !ok {
+		return errMissingMetadata
+	}
+	if !valid(md["authorization"]) {
+		return errInvalidToken
+	}
+
+	err := handler(srv, newWrappedStream(ss.Context(), ss))
+	if err != nil {
+		logger("RPC failed with error %v", err)
+	}
+	return err
+}
+
+func injectResourceAccessUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+	// Assuming authentication was successful in the middleware chain, fetch the subject access, mocked here
+	mdCtx := metadata.NewIncomingContext(ctx, mockedMetadata)
+	return handler(mdCtx, req)
+}
+
+func injectResourceAccessStreamInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	// Assuming authentication was successful in the middleware chain, fetch the subject access, mocked here
+	mdCtx := metadata.NewIncomingContext(ss.Context(), mockedMetadata)
+	return handler(srv, newWrappedStream(mdCtx, ss))
+}
+
+func main() {
+	flag.Parse()
+
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", *port))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+
+	// Create tls based credential.
+	creds, err := credentials.NewServerTLSFromFile(data.Path("x509/server_cert.pem"), data.Path("x509/server_key.pem"))
+	if err != nil {
+		log.Fatalf("failed to create credentials: %v", err)
+	}
+
+	staticInteceptor, err := authz.NewStatic(authzPolicy)
+	if err != nil {
+		log.Fatalf("failed to create static authz interceptor: %v", err)
+	}
+	s := grpc.NewServer(
+		grpc.Creds(creds),
+		grpc.ChainUnaryInterceptor(
+			authUnaryInterceptor,
+			injectResourceAccessUnaryInterceptor,
+			staticInteceptor.UnaryInterceptor,
+		),
+		grpc.ChainStreamInterceptor(
+			authStreamInterceptor,
+			injectResourceAccessStreamInterceptor,
+			staticInteceptor.StreamInterceptor,
+		),
+	)
```

Comment:
Please consider rewriting this as:

```golang
unaryInt := grpc.ChainUnaryInterceptor(authUnaryInterceptor, injectResourceAccessUnaryInterceptor, staticInteceptor.UnaryInterceptor),
streamInt := grpc.ChainStreamInterceptor(authStreamInterceptor, injectResourceAccessStreamInterceptor, staticInteceptor.StreamInterceptor)
s := grpc.NewServer(grpc.Creds(creds), unaryInt, streamInt)
```

Please see: https://google.github.io/styleguide/go/guide#line-length

Reply:
Refactored